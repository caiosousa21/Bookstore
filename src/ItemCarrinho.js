import React, { Component } from 'react';

class ItemCarrinho extends Component {
    render() {
        return (
            <div className='ItemCarrinho'>
                <p className='PItem'>{this.props.item}</p>
                <button className='ExcluirItem' onClick={()=>{this.props.clickExcluir(this.props.index)}}></button>
                <hr></hr>
            </div>
        )
    }
}
export default ItemCarrinho;


