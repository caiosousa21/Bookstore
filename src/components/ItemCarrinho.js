import React from 'react';

function ItemCarrinho(props) {
    return (
        <div className='ItemCarrinho'>
            <p className='PItem'>{props.item}</p>
            <button className='ExcluirItem' onClick={()=>props.clickExcluir(props.item)}></button>
            <hr></hr>
        </div>
    )

}
export default ItemCarrinho;