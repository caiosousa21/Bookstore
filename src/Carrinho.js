import React, { Component } from 'react';
import ItemCarrinho from './ItemCarrinho';

class Carrinho extends Component {
    carregarItem = () => {
        let item = [];
        for (let i = 0; i < this.props.interior.length; i++) {
            item.push(<ItemCarrinho key={'item'+i} item={this.props.interior[i]} clickExcluir={this.props.clickExcluir} index={i}/>);
        }
        return item
    }

    render() {
        return (
            <div className='Carrinho'>
                {this.carregarItem()}
            </div>
        )
    }
}

export default Carrinho;