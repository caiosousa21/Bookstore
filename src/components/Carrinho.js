import React from 'react';
import ItemCarrinho from './ItemCarrinho';
import id from 'uuid/v1'

function Carrinho(props){
    
        return (
            <div className='Carrinho'>
                {props.carrinho.map(
                    item => 
                        <ItemCarrinho item={item} key={id} clickExcluir={props.clickExcluir}/>     
                    )
                }
            </div>
        )
    
}



export default Carrinho;