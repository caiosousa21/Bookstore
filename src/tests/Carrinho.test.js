import React from 'react';
import Carrinho from '../components/Carrinho';
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import ItemCarrinho from '../components/ItemCarrinho'

Enzyme.configure({ adapter: new Adapter() })

describe('Carrinho', () => {
    test('Renderizando Carrinho', () => {
        const wrapper = shallow(<Carrinho interior={'x'}/>);
        expect(wrapper.exists()).toBe(true);
    }),
    it('Testando interface com snapshots',()=>{
        const tree = renderer.create(<Carrinho interior={''}/>).toJSON();
        expect(tree).toMatchSnapshot();
        const tree2 = renderer.create(<Carrinho interior={['caio','neymar','zé']}/>).toJSON();
        expect(tree2).toMatchSnapshot();
    }),
    it('Testando criação de itemCarrinho manualmente',()=>{
        const wrapper = shallow(<Carrinho interior={'x'}/>);
        const foo = wrapper.instance().carregarItem();
    expect(JSON.stringify(foo)).toEqual(JSON.stringify([<ItemCarrinho key='item0' item='x' index={0}/>]))
    })
})