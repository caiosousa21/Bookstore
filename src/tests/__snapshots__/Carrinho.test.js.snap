// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carrinho Testando interface com snapshots 1`] = `
<div
  className="Carrinho"
/>
`;

exports[`Carrinho Testando interface com snapshots 2`] = `
<div
  className="Carrinho"
>
  <div
    className="ItemCarrinho"
  >
    <p
      className="PItem"
    >
      caio
    </p>
    <button
      className="ExcluirItem"
      onClick={[Function]}
    />
    <hr />
  </div>
  <div
    className="ItemCarrinho"
  >
    <p
      className="PItem"
    >
      neymar
    </p>
    <button
      className="ExcluirItem"
      onClick={[Function]}
    />
    <hr />
  </div>
  <div
    className="ItemCarrinho"
  >
    <p
      className="PItem"
    >
      z√©
    </p>
    <button
      className="ExcluirItem"
      onClick={[Function]}
    />
    <hr />
  </div>
</div>
`;
